### DISCUSSION TITLE
CORSMiddleware with Access-Control-Allow-Private-Network header

### DISCUSSION THREAD

**Discussion by Impro02** [fastapi/fastapi#11145] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# FASTAPI
from fastapi.middleware.cors import CORSMiddleware as _CORSMiddleware

# STARLETTE
from starlette.types import ASGIApp

class CORSMiddleware(_CORSMiddleware):
    def __init__(
        self,
        app: ASGIApp,
        allow_origins: Sequence[str] = (),
        allow_methods: Sequence[str] = ("GET",),
        allow_headers: Sequence[str] = (),
        allow_credentials: bool = False,
        allow_private_network: bool = False,
        allow_origin_regex: Optional[str] = None,
        expose_headers: Sequence[str] = (),
        max_age: int = 600,
    ) -> None:
        super().__init__(
            app,
            allow_origins,
            allow_methods,
            allow_headers,
            allow_credentials,
            allow_origin_regex,
            expose_headers,
            max_age,
        )

        if allow_private_network:
            self.simple_headers[
                "Access-Control-Allow-Private-Network"
            ] = "true"
```

### Description

Is there a reason why the Access-Control-Allow-Private-Network header cannot be added from the CORSMiddleware class ?
Is it possible to add a parameter allow_private_network like the example ?

### Operating System

Linux, Windows

### FastAPI Version

0.109.2

### Pydantic Version

2.6

### Python Version

3.12.1

**Comment by Kludex** [comment, marked as answer]
This should be in Starlette.

**Reply by Impro02** [reply]
Why in starlette and not in the fastapi  CORSMiddleware class like Access-Control-Allow-Credentials ?

**Reply by Kludex** [reply]
Because FastAPI doesn't maintain the CORSMiddleware: https://github.com/tiangolo/fastapi/blob/master/fastapi/middleware/cors.py

**Comment by Impro02** [comment]
The discussion on starlette: https://github.com/encode/starlette/discussions/2509
